--- a/hook.c
+++ b/hook.c
@@ -783,59 +783,11 @@
 int run_hooks_opt(struct repository *r, const char *hook_name,
 		  struct run_hooks_opt *options)
 {
-	struct string_list *hook_list = list_hooks(r, hook_name, options);
-	struct hook_cb_data cb_data = {
-		.rc = 0,
-		.hook_name = hook_name,
-		.hook_command_list = hook_list,
-		.options = options,
-	};
-	int ret = 0;
-	unsigned int jobs = get_hook_jobs(r, options, hook_name, hook_list);
-	const struct run_process_parallel_opts opts = {
-		.tr2_category = "hook",
-		.tr2_label = hook_name,
-
-		.processes = jobs,
-		.ungroup = jobs == 1,
-
-		.get_next_task = pick_next_hook,
-		.start_failure = notify_start_failure,
-		.feed_pipe = options->feed_pipe,
-		.task_finished = notify_hook_finished,
-
-		.data = &cb_data,
-	};
-
-	if (!options)
-		BUG("a struct run_hooks_opt must be provided to run_hooks");
-
-	if (options->path_to_stdin && options->feed_pipe)
-		BUG("options path_to_stdin and feed_pipe are mutually exclusive");
-
-	/*
-	 * Ensure cb_data copy and free functions are either provided together,
-	 * or neither one is provided.
-	 */
-	if (!options->feed_pipe_cb_data_alloc != !options->feed_pipe_cb_data_free)
-		BUG("feed_pipe_cb_data_alloc and feed_pipe_cb_data_free must be set together");
-
-	if (options->invoked_hook)
+	/* safer-git: all hooks permanently disabled at build time */
+	if (options && options->invoked_hook)
 		*options->invoked_hook = 0;
-
-	if (!cb_data.hook_command_list->nr) {
-		if (options->error_if_missing)
-			ret = error("cannot find a hook named %s", hook_name);
-		goto cleanup;
-	}
-
-	run_processes_parallel(&opts);
-	ret = cb_data.rc;
-cleanup:
-	string_list_clear_func(cb_data.hook_command_list, hook_free);
-	free(cb_data.hook_command_list);
 	run_hooks_opt_clear(options);
-	return ret;
+	return 0;
 }
 
 int run_hooks(struct repository *r, const char *hook_name)
